browser/src/types.d.ts

```diff
@@ -14,6 +35,83 @@ export type GeneMetadata = {
   flags: string[]
 }
 
+export type ExacConstraint = {
+  exp_syn: number
+  obs_syn: number
+  syn_z: number
+  exp_mis: number
+  obs_mis: number
+  mis_z: number
+  exp_lof: number
+  obs_lof: number
+  pLI: number
+}
+
+export type GnomadConstraint = {
+  exp_lof: number
+  exp_mis: number
+  exp_syn: number
+  obs_lof?: number
+  obs_mis?: number
+  obs_syn?: number
+  oe_lof: number
+  oe_lof_lower: number
+  oe_lof_upper: number
+  oe_mis: number
+  oe_mis_lower: number
+  oe_mis_upper: number
+  oe_syn: number
+  oe_syn_lower: number
+  oe_syn_upper: number
+  lof_z?: number
+  mis_z: number
+  syn_z: number
+  pLI: number
+  flags?: string[]
+}
+
+export type VariantCooccurrenceCountsPerSeverityAndAf<
+  Severities,
+  AfCutoffs,
+  CountCellSchema
+> = Record<Severities, Record<AfCutoffs, CountCellSchema>>
+
+export type HeterozygousVariantCooccurrenceAfCutoff = typeof heterozygousVariantCooccurrenceAfCutoffs[number]
+
+export type HeterozygousVariantCooccurrenceSeverity = typeof heterozygousVariantCooccurrenceSeverities[number]
+
+export type HeterozygousCountCellSchema = {
+  in_cis: number
+  in_trans: number
+  unphased: number
+  two_het_total: number
+}
+
```

Comment:
Continued discussion from office hours, this may be a good time to start breaking these type definitions into slightly more scoped files. A file like this in the Gene page directory with only the types that are relevant there might help with potential name collisions, and slow the rapid growth of this global types file.

Continuing to add to `types.d.ts` seems like it's going to cause problems sooner rather than later. Though I know there are some types that are more widespread than just in Genes, some amount of tighter scoping may start to be useful.

Reply:
@mattsolo1 FYI what Riley refers to here is that I said to him earlier that in retrospect a centralized `types.d` was probably a mistake, and it would be better to generally have types live in the code that they're most relevant to.